libkbfs/disk_block_cache_service.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"context"
+
+	"github.com/keybase/kbfs/kbfsblock"
+	"github.com/keybase/kbfs/kbfscrypto"
+	kbgitkbfs "github.com/keybase/kbfs/protocol/kbgitkbfs1"
+	"github.com/keybase/kbfs/tlf"
+)
+
+type diskBlockCacheServiceConfig interface {
+	diskBlockCacheGetter
+}
+
+// DiskBlockCacheService delegates requests for blocks to this KBFS
+// instance's disk cache.
+type DiskBlockCacheService struct {
+	config diskBlockCacheServiceConfig
+}
+
+var _ kbgitkbfs.DiskBlockCacheInterface = (*DiskBlockCacheService)(nil)
+
+// NewDiskBlockCacheService creates a new DiskBlockCacheService.
+func NewDiskBlockCacheService(config diskBlockCacheServiceConfig) *DiskBlockCacheService {
+	return &DiskBlockCacheService{
+		config: config,
+	}
+}
+
+// GetBlock implements the DiskBlockCacheInterface interface for
+// DiskBlockCacheService.
+func (cache *DiskBlockCacheService) GetBlock(ctx context.Context,
+	arg kbgitkbfs.GetBlockArg) (kbgitkbfs.GetBlockRes, error) {
+	dbc := cache.config.DiskBlockCache()
```

Comment:
What if this one happens to be a remote one too? Do we need to have a check and bail out if that's the case? I guess it's not possible for now since the GRH doesn't serve this service.

Reply:
IMO it's fine for this to work.

Reply:
I was worried we'd have circular dependency if RPC topology gets more complicated in the future. But that's too early to worry about I guess.

Reply:
Yeah, I'll add a comment though so we pay attention to it.